packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -441,7 +447,12 @@ contract OPContractsManager is ISemver {
 
         Implementations memory impls = thisOPCM.implementations();
         Blueprints memory bps = thisOPCM.blueprints();
-        // TODO: upgrading the SuperchainConfig and ProtocolVersions (in a new function)
+
+        if (address(_superchainProxyAdmin) != address(0)) {
+            // Attempt to upgrade. If the ProxyAdmin is not the SuperchainConfig's admin, this will revert.
+            upgradeTo(_superchainProxyAdmin, address(superchainConfig), impls.superchainConfigImpl);
+            upgradeTo(_superchainProxyAdmin, address(protocolVersions), impls.protocolVersionsImpl);
+        }
```

Comment:
Instead of a sentinel value for when to skip this upgrade, can we just compare this OPCM's implementation to the current one? I think the answer is no because of the `proxyCallIfNotAdmin` modifier, so there's no way to get the impl address onchain?
```suggestion
        if (impls.superchainConfigImpl != superchainConfig.implementation()) {
          upgradeTo(_superchainProxyAdmin, address(superchainConfig), impls.superchainConfigImpl);
        }
        if (impls.protocolVersionsImpl != protocolVersions.implementation()) {
          upgradeTo(_superchainProxyAdmin, address(protocolVersions), impls.protocolVersionsImpl);
        }
```

Reply:
I think the closest we can get to determining if a contract has been upgraded or not is by reading the semvers.

One side effect of doing that though, would be that it's impossible to use `opcm.upgrade()` before the `upgrader` has upgraded them. That might be desirable, but if so we should implement that more directly.